scripts/run_allocation_algorithms.py

```diff
@@ -51,19 +46,18 @@
 
 # construct schedule
 schedule = []
-topic_map = defaultdict(set)
+topic_map = defaultdict(list)
 for idx, (_, row) in enumerate(df.iterrows()):
     crs = str(row["Catalog"])
-    topic_map[row["Categories"]].add(crs)
     slt = slots_for_time_range(row["Mtg Time"], slot.times)
     sec = row["Section"]
     capacity = row["CICScapacity"]
     dys = tuple([day.strip() for day in row["zc.days"].split(" ")])
-    schedule.append(
-        ScheduleItem(features, [crs, slt, dys, sec], index=idx, capacity=capacity)
-    )
+    item = ScheduleItem(features, [crs, slt, dys, sec], index=idx, capacity=capacity)
+    schedule.append(item)
+    topic_map[row["Categories"]].append(item)
 
-topics = sorted([sorted(list(courses)) for courses in topic_map.values()])
+topics = [topic for topic in topic_map.values()]
```

Comment:
My understanding here is that you are changing topics to be lists of ScheduleItems instead of sets of the course strings? 
All other changes in the run_allocation_algorithms.py and in survey_simulation.py are mostly cosmetic, right?

Reply:
Yes, that's right.